src/electron/process_manager.ts

```diff
@@ -366,6 +384,18 @@ function checkUdpForwardingEnabled() {
   });
 }
 
+// Checks whether UDP forwarding support has changed and restarts tun2socks if it has.
+async function handleUdpSupportChange(onDisconnected: () => void): Promise<void> {
+  const isUdpSupported = await checkUdpForwardingEnabled().then(() => true).catch(e => false);
```

Comment:
In two places `checkUdpForwardingEnabled` is used to compute a boolean - what do you think of making it return/resolve with true/false?

Reply:
I thought about it, but I concluded that it would uncecessarily complicate the logic in the caller, as it would have to handle failures, which are currently treated as "no UDP support". Happy to make the change if you think it will improve readability.

Reply:
I think you're right. I made the change in 3028380. Thanks for the suggestion!